File: soma de divisores/soma de divisores.py
x=1
contador=0
acumulador=0
t=0
while x<=5:
    num=int(input())
    for c in range(1,num+1):
        if num%c==0:
            acumulador+=c
    if acumulador>t:
        contador=num
    t=acumulador
    acumulador=0
    x+=1
print(contador)

File: soma de divisores/Soma de Divisores.c
#include <stdio.h>

int main(){
  int numero,acumulador = 0, contador = 0, resultado, i, x;

  for (i = 0; i < 5; i++){
    scanf("%d", &numero);
    for(x = 1; x <= numero; x++){
      if(x % numero == 0){
        acumulador+= x;
      }
    }
    if (acumulador > contador){
      resultado = numero;
    }
    contador = acumulador;
    acumulador = 0;
  }
  printf("%d", resultado);
  return 0;
}